scrape/scrape.go

```diff
@@ -1277,33 +1285,29 @@ func (sl *scrapeLoop) setScrapeFailureLogger(l *logging.JSONFileLogger) {
 }
 
 func (sl *scrapeLoop) run(errc chan<- error) {
-	if !sl.skipOffsetting {
-		select {
-		case <-time.After(sl.scraper.offset(sl.interval, sl.offsetSeed)):
-			// Continue after a scraping offset.
-		case <-sl.ctx.Done():
-			close(sl.stopped)
-			return
-		}
+	jitterDelayTime := sl.scraper.offset(sl.interval, sl.offsetSeed)
+	if sl.initialScrapeOffset != nil {
+		jitterDelayTime = *sl.initialScrapeOffset
+	}
```

Comment:
Here is one the few moments where I actually like using an `else`

`scrapeLoop.run` is a hot path I believe, if `sl.initialScrapeOffset` is not nil there's no reason to measure the jitterDelay just to override it later.

```go
var delayTime time.Duration
if sl.initialScrapeOffset != nil {
	delayTime = *sl.initialScrapeOffset
} else {
	delayTime := sl.scraper.offset(sl.interval, sl.offsetSeed)
}
```